### DISCUSSION TITLE
Not able to send logs to wazuh

### DISCUSSION THREAD

**Discussion by EvaSzlifierzCriado** [wazuh/wazuh#16258] [Q&A]
Hello community,

I want to send logs from a fortinet firewall and visualize them on a Wazuh cluster. This is my configuration for the fortinet firewall:

```
config log syslogd3 setting
    set status enable
    set server "my_domain"
    set mode reliable
    set port 5144
    set facility local7
    set source-ip ''
    set format default
    set priority default
    set max-log-rate 0
    set enc-algorithm disable
    set interface-select-method auto
end
```

And this is my configuration for logstash in wazuh:

```
input {
    tcp {
        port => 5144
    }
}
filter {
}
output {
    stdout {
        codec => json
    }
}
```

When I try to send logs to a server I can visualize them properly, both with tcp and udp, but I can't if I send them to wazuh. How can I solve this? I don't understand what I am doing wrong. Thank you in advance :)

**Comment by jctello** [comment]
Hi @evaa1234 ,
In order to have Wazuh analyze the logs you can configure [Remote syslog collection](https://documentation.wazuh.com/current/user-manual/capabilities/log-data-collection/how-it-works.html#remote-syslog) on the Wazuh managers by adding to its configuration a stanza such as the following:
```
<ossec_config>
  <remote>
    <connection>syslog</connection>
    <port>5144</port>
    <protocol>udp</protocol>
    <allowed-ips>0.0.0.0/0</allowed-ips>
  </remote>
</ossec_config>
```
This way the information is collected and analyzed alongside all other security events collected by the Wazuh platform.

If due to the nature of your network's topography you prefer to collect syslog using an endpoint instead of the Wazuh manager you may use rsyslog or logstash to collect the messages and place them onto a file which should then be collected by the Wazuh agent as explained here https://documentation.wazuh.com/current/cloud-service/your-environment/send-syslog-data.html

Things to consider for further troubleshooting:
- By default Wazuh logs only events that are identified to be security relevant, that is, that match a rule level 3 or above. You may need to create custom ruleset content to identify messages which aren't identified by Wazuh out of the box: https://documentation.wazuh.com/current/user-manual/ruleset/custom.html
- By enabling [<logall-json>](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) on the Wazuh manager you may determine if events are being received by the wazuh manager and use the `full_log` field to test the ruleset, verify which rules they trigger and help you in creating your own rules.
- You may determine using `tcpdump -i any port 5144` if logs are reaching the machine collecting the logs. But take into consideration that tcpdump will see messages before they go through the operating system's firewall so they may still be blocked by `firewalld` or `SELinux`

**Reply by EvaSzlifierzCriado** [reply]
If I establish the same port on the config file on logstash and on the ossec config remote (the port 514 in my case), it says that tha address is already in use, even though it is a port which is not used by any other application. Why is this happening?

Also thanks a lot for the links, I didn't know you have to configure specifically the rsyslog file.

**Reply by jctello** [reply]
In order to capture messages sent to a port Wazuh needs to bind as listening to it, this cannot happen if another program, in this case Logstash is already bound to the same port.  So in order to proceed you must first stop Logstash or modify the configuration and restart the service.

**Reply by EvaSzlifierzCriado** [reply]
Ok, so if I want to receive the logs on Wazuh I must configure syslog on the ossec config as you explained, and if I just want the pod receives the logs and view them on terminal or a file for instance, I should use rsyslog or logstash, right?
Also, how can I check that the logs arriving at my wazuh are in fact the ones on that firewall? Since I receive logs from other devices, is there a way to check that a log is from that specific device?

**Reply by jctello** [reply]
Yes, you may use Wazuh to collect and analyze the syslog messages through the `ossec.conf` file but if you're not interested in analyzing this information you may collect them with another tool.

As for uniquely identifying each log source, when collecting logs directly on the Wazuh manager via syslog, the resulting alerts will have the IP of the source in the `location` field. Also if the network device is following the syslog standard format then Wazuh will be able to extract its hostname in the `predecoded.hostname` field.
